needy/filesystem.py

```diff
@@ -0,0 +1,26 @@
+import os
+import time
+import fcntl
+
+from datetime import datetime, timedelta
+
+
+def lock_file(file_path, timeout=None):
+    '''returns file descriptor to newly locked file or None if file couldn't be locked'''
+    fd = os.open(file_path, os.O_RDWR | os.O_CREAT)
+    if timeout is None:
+        fcntl.flock(fd, fcntl.LOCK_EX)
+        return fd
+    start = datetime.now()
+    lock_acquired = False
+    while True:
+        try:
+            fcntl.flock(fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
+            lock_acquired = True
+        except IOError:
+            pass
+        if lock_acquired or datetime.now() > start + timedelta(seconds=timeout):
+            break
+        time.sleep(1)
```

Comment:
I can work around it, but it's worth mentioning that the suggested method is somewhat global to the process even if it is better.